scout/parse/config_base.py

```diff
@@ -0,0 +1,262 @@
+    vcf_str: Optional[str] = None
+    vcf_sv: Optional[str] = None
+    vcf_sv_research: Optional[str] = None
+
+
+class ScoutLoadConfig(BaseModel):
+    analysis_date: Any = datetime.datetime.now()
+    assignee: str = None  ## ??
+    case_id: str = Field(..., alias="family")
+    #  chromograph_image_files: Optional[List[str]]
+    cnv_report: Optional[str] = None
+    cohorts: Optional[List[str]] = None
+    collaborators: Optional[List[str]] = None
+    coverage_qc_report: str = None  ## ??
+    default_panels: Optional[List[str]] = Field([], alias="default_gene_panels")
+    delivery_report: Optional[str] = None
+    display_name: str = None
+    family: str = None
+    family_name: Optional[str] = None
+    gene_panels: Optional[List[str]] = []
+    gene_fusion_report: Optional[str] = None
+    gene_fusion_report_research: Optional[str] = None
+    genome_build: int = Field([], alias="human_genome_build")
+    human_genome_build: str = None
+    individuals: List[ScoutIndividual] = Field([], alias="samples")
+    lims_id: Optional[str] = None
+    madeline_info: Optional[str] = Field("", alias="madeline")
+    multiqc: Optional[str] = None
+    owner: str = None
+    peddy_check: Optional[str] = None
+    peddy_ped: Optional[str] = None
+    peddy_sex: Optional[str] = None
+    phenotype_terms: Optional[List[str]] = None
+    rank_model_version: Optional[str] = ""
+    rank_score_threshold: int = 0
+    samples: List[ScoutIndividual] = []
+    smn_tsv: Optional[str] = None
+    sv_rank_model_version: Optional[str] = ""
+    synopsis: Union[List[str], str] = None
+    track: Literal["rare", "cancer"] = "rare"
+    vcf_files: Optional[VcfFiles] = None
+
+    # override init() for handling nested vcf_files dicts
+    # use try/except to handle TypeError if `vcf_files`is already set in
+    # previous call `parse_case_data()`, `parse_case()`.
+    def __init__(self, **data):
+        vcfs = VcfFiles(**data)
+        try:
+            super().__init__(vcf_files=vcfs, **data)
+        except TypeError as err:
+            super().__init__(**data)
+
+    @validator("analysis_date")
+    def check_analysis_date(cls, dt):
+        if isinstance(dt, datetime.datetime):
+            # return datetime.datetime.strptime(dt, "%Y-%M-%d %H:%M%S")
+            return dt
+        correct_date = datetime.datetime.now()
+        return correct_date
+
+    # Each case has to have a owner. If not provided in config file it needs to be given as a
+    # argument
+    @validator("owner", pre=True, always=True)
+    def mandatory_check_owner(cls, value):
+        if value is None:
+            raise ConfigError("A case has to have a owner")
+        return value
+
+    @validator("family", pre=True, always=True)
+    def mandatory_check_family(cls, value):
+        if value is None:
+            raise ConfigError("A case has to have a 'family'")
+        return value
+
+    @validator("madeline_info")
+    def check_if_madeline_exists(cls, path):
```

Comment:
Doesn't look like this one is working. The pedigree is missing in the demo case.. 🤔